pkg/provider/azure/keyvault/keyvault_test.go

```diff
@@ -483,6 +485,42 @@ func makeValidFind() *esv1beta1.ExternalSecretFind {
 	}
 }
 
+func getStore(testName string) *esv1beta1.SecretStore {
+	if testName == "specIsNil" {
+		store := &esv1beta1.SecretStore{}
+		return store
+	}
+	if testName == "providerIsNil" {
+		store := &esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{},
+		}
+		return store
+	}
+	if testName == "azureKVIsNil" {
+		store := &esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{},
+			},
+		}
+		return store
+	}
+	return nil
+}
+
+func getValidateStore(t *testing.T, testName string) {
+	a := &Azure{}
+	if testName == "storeIsNil" {
+		if err := a.ValidateStore(nil); (err != nil) != true {
+			t.Errorf(errStore, err, true)
+		}
+	} else {
+		store := getStore(testName)
+		if err := a.ValidateStore(store); (err != nil) != true {
+			t.Errorf(errStore, err, true)
+		}
+	}
+}
+
 func TestValidateStore(t *testing.T) {
 	type args struct {
 		auth esv1beta1.AzureKVAuth
```

Comment:
If you change the type signature here to `*esv1beta1.SecretStore` you can define one individual SecretStore per test case and avoid the complexity